# OLD CODE

In [ ]:
# Open the dc.tif file and the original Virginia Beach GeoTIFF file
with rasterio.open(dc_tif_path) as dc_src, rasterio.open(vb_tif_path) as vb_src:
    # Read the metadata of each image
    dc_metadata = dc_src.meta
    vb_metadata = vb_src.meta
    
    # Update the Virginia Beach metadata to match DC metadata, except for width, height, and count
    vb_metadata.update({
        key: value for key, value in dc_metadata.items() if key not in ['width', 'height', 'count']
    })
    
    # Save the updated Virginia Beach GeoTIFF file with the new metadata
    with rasterio.open(
        updated_vb_tif_path,
        'w',
        **vb_metadata
    ) as dst:
        for i in range(1, vb_src.count + 1):
            dst.write(vb_src.read(i), i)
    
    # Plot the metadata of each image side by side
    fig, axes = plt.subplots(1, 2, figsize=(15, 7))
    
    axes[0].imshow(dc_src.read(1), cmap='gray')
    axes[0].set_title('First Band of DC GeoTIFF Image')
    axes[0].axis('off')
    
    axes[1].imshow(vb_src.read(1), cmap='gray')
    axes[1].set_title('First Band of Virginia Beach GeoTIFF Image')
    axes[1].axis('off')
    
    plt.show()
    
    # Print metadata
    print("DC GeoTIFF Metadata:")
    for key, value in dc_metadata.items():
        print(f"{key}: {value}")
    
    print("\nVirginia Beach GeoTIFF Metadata (Updated):")
    for key, value in vb_metadata.items():
        print(f"{key}: {value}")

In [ ]:
"""
- Common colormaps include: 'viridis', 'plasma', 'inferno', 'magma', 'cividis', 'gray', 'hot', 'cool', 'spring', 'summer', 'autumn', 'winter'.
"""
def plot_hyperspectral_images_with_keys(path, keys, band=0):
    # Load the .mat file
    data = scipy.io.loadmat(path)
    
    # Extract the 3D arrays using the provided keys
    arrays = [data[key] for key in keys]
    
    # Extract the specified band (assuming the third dimension is the spectral dimension)
    bands = [array[:, :, band] for array in arrays]
    
    # Plot the images side by side
    fig, axes = plt.subplots(1, len(bands), figsize=(15, 5))
    
    for ax, band, key in zip(axes, bands, keys):
        ax.imshow(band, cmap='viridis')
        ax.imshow(band, cmap='viridis', alpha=0)
        ax.set_title(f'{key}')
        ax.axis('off')
    plt.show()
    
plot_hyperspectral_images_with_keys(
    './data/HSI_Data/Hyperspectral_Project/WDC/results/hypersigma/hypersigma_Case5.mat',
    keys=['gt', 'input', 'output'],
    band=60  # Change this value to plot a different band
)

In [ ]:
wdc_test = './data/HSI_Data/Hyperspectral_Project/WDC/test/test.mat'
hypso_test = './data/HSI_Data/Hyperspectral_Project/HYPSO/test/test.mat'
wdc_test_case='./data/HSI_Data/Hyperspectral_Project/WDC/test_noise/Cases/Case2/test.mat'
hypso_test_case='./data/HSI_Data/Hyperspectral_Project/HYPSO/test_noise/Cases/Case2/test.mat'
# Load the .mat files
wdc_data = scipy.io.loadmat(wdc_test)
hypso_data = scipy.io.loadmat(hypso_test)
wdc_case_data = scipy.io.loadmat(wdc_test_case)
hypso_case_data = scipy.io.loadmat(hypso_test_case)

# BEGIN: shape of wdc case data and hypso case data
wdc_case_shape = wdc_case_data['gt'].shape
hypso_case_shape = hypso_case_data['gt'].shape

print("Shape of WDC case data:", wdc_case_shape)
print("Shape of HYPSO case data:", hypso_case_shape)
# END:

Shape of WDC case data: (191, 192, 192)
Shape of HYPSO case data: (120, 192, 192)


create wdc...
Map size(GB): 0.23591503500938416
loading mat: train_0.mat
-------------------
Preprocessing data...
Preprocess data shape: (191, 600, 307)
zooming to 1.000000
zooming to 0.500000
zooming to 0.250000
new data 0 shape  (544, 191, 64, 64)
new data 1 shape  (360, 191, 64, 64)
new data 2 shape  (22, 191, 64, 64)
concatenating data...
new data shape after conc: (926, 191, 64, 64)
augmenting data...
Postprocess data shape: (926, 191, 64, 64)
/home/lofty/CODE/HyperSIGMA-fork/ImageDenoising/data/HSI_Data/Hyperspectral_Project/WDC/wdc.db
load mat: train_0.mat
-------------------
Preprocessing data...
Preprocess data shape: (191, 600, 307)
zooming to 1.000000
zooming to 0.500000
zooming to 0.250000
new data 0 shape  (544, 191, 64, 64)
new data 1 shape  (360, 191, 64, 64)
new data 2 shape  (22, 191, 64, 64)
concatenating data...
new data shape after conc: (926, 191, 64, 64)
augmenting data...
Postprocess data shape: (926, 191, 64, 64)
load mat: train_1.mat
-------------------
Preprocessing data...
Preprocess data shape: (191, 480, 307)
zooming to 1.000000
zooming to 0.500000
zooming to 0.250000
new data 0 shape  (432, 191, 64, 64)
new data 1 shape  (276, 191, 64, 64)
new data 2 shape  (16, 191, 64, 64)
concatenating data...
new data shape after conc: (724, 191, 64, 64)
augmenting data...
Postprocess data shape: (724, 191, 64, 64)
done